addons/mail/models/discuss/discuss_channel.py

```diff
@@ -689,27 +688,27 @@ def _notify_get_recipients(self, message, msg_vals=False, **kwargs):
                     'ushare': ushare,
                 })
 
-        domain = expression.AND([
+        domain = Domain.AND([
```

Comment:
How do you decide when using `Domain.AND`/`OR` as opposed to using `&` or `|` operators?

It feels like using `AND`/`OR` really makes sense when the value is a _variable_ that should be iterated. But not when there is a static number of params like here.

Did you avoid changing this one just because it was longer than some others?
If so, wouldn't the minimal diff in these PR be always with a direct replace of `expression` by `Domain`?

Reply:
Usually, I use `Domain.AND` if there is more than 2-3 conditions to combine or a variable amount (`Domain.AND(xxx for x in ...)`). I also try to avoid changing too many lines if there is no need.

A simple replace would work, but it is sometimes much longer: the most common example is:
`d = expression.AND([d, d2])` which becomes `d &= d2`.